martus-client/source/org/martus/client/test/TestSha1ToSha2Change.java

```diff
@@ -157,6 +159,76 @@ public void testLoadTemplateManagerStateSignedWithSha2UsingSha1Verifier() throws
 		}
 	}
 
+	public void testReadBulletinSignedWithSha1UsingSha2Verifier() throws Exception
+	{
+		MockBulletinStore testStore = new MockBulletinStore(securitySha1);
+		assertEquals(0, testStore.getBulletinCount());
+
+		final String initialSummary = "New bulletin";
+
+		Bulletin bulletin = testStore.createEmptyBulletin();
+		bulletin.set(Bulletin.TAGSUMMARY, initialSummary);
+
+		testStore.saveBulletin(bulletin);
+		assertEquals(1, testStore.getBulletinCount());
+
+		testStore = changeBulletinStoreSecurity(testStore, securitySha2);
+		UniversalId uId = bulletin.getUniversalId();
+
+		bulletin = testStore.getBulletinRevision(uId);
+		assertEquals("not saved initially?", initialSummary, bulletin.get(Bulletin.TAGSUMMARY));
+	}
+
+	public void testReadBulletinSignedWithSha2UsingSha1Verifier() throws Exception
+	{
+		MockBulletinStore testStore = new MockBulletinStore(securitySha2);
+		assertEquals(0, testStore.getBulletinCount());
+
+		Bulletin bulletin = testStore.createEmptyBulletin();
+
+		testStore.saveBulletin(bulletin);
+		assertEquals(1, testStore.getBulletinCount());
+
+		testStore = changeBulletinStoreSecurity(testStore, securitySha1);
+		UniversalId uId = bulletin.getUniversalId();
+		assertNull(testStore.getBulletinRevision(uId));
```

Comment:
The assert null should have a message to state why we were not expecting a null.